### ISSUE TITLE
BIBFRAME in Sinopia issue: Admin metadata: Cataloger ID does not display in the Sinopia RT.

### ISSUE THREAD

**Issue by gerontakos** [uwlib-cams/rml#32]
Cataloger ID property displays in the Sinopia RT but the value does not. It appears to be correctly display in the json-ld. True for BF Items, Instances, and Works.

**Comment by CECSpecialistI** [comment]
I can't find the source of this problem. We have a number of RT's in Sinopia for the same bf:Class here. Which RT are we using? Should we request that all but one be deleted @mcm104 ?

**Comment by mcm104** [comment]
I'm not sure I understand your comment fully -- what RT are we using for what? On my end, I'm only selecting WAU:RT:BF2:Work, :Instance, or :Item for each resource -- beyond that, nodes are matched to their RTs within Sinopia and I have no control over it.

Also unsure what you mean by all but one being deleted -- all RTs? Then add them back in? Do you think that would work? I honestly have no idea.

My thought is that if nothing looks wrong within the admin metadata RT, I would think we would need to raise an issue with Sinopia devs.

**Comment by CECSpecialistI** [comment]
There are a few admin metadata RT's we have used, I'm thinking these errors are potentially stemming from using one of the old ones in the RML code? What is the ID for the AdminMetadata resource template we're using here?

**Comment by mcm104** [comment]
The conversion code isn't selecting an AdminMetadata RT -- it only selects a Work, Instance, or Item RT. Whatever AdminMetadata RT is nested within the :Works, :Instance, or :Item RTs is what gets used in Sinopia, which looks like WAU:RT:BF2:AdminMetadata.

I'm also not seeing anything wrong with the admin metadata RT or with how it's nested in the Work, Instance, and Item RTs, so I still think it's a dev problem.

This might be clearer to me since I'm the one writing the conversion code, so as an example of what I'm doing on my end, with [this example](https://api.sinopia.io/resource/0c649743-5192-4a5f-b195-b4fff10d5757):

```
{
  "data": [ ... ],
  "user": "mcm104",
  "group": "washington",
  "templateId": "WAU:RT:BF2:Works",
  "types": [
    "http://id.loc.gov/ontologies/bibframe/Work"
  ],
  "id": "0c649743-5192-4a5f-b195-b4fff10d5757",
  "uri": "https://api.sinopia.io/resource/0c649743-5192-4a5f-b195-b4fff10d5757",
  "timestamp": "2021-06-01T22:44:18.881Z"
}
```

(I collapsed the "data" section so it's easier to see the overall structure of the JSON-LD as it appears in Sinopia.)

Everything within "data" is what the RML code creates -- this is where all the values from our RDA end up. Aside from things like language tags or datatypes, I'm not adding or editing anything in there. The admin metadata in there is taken from our RDA using a one-to-one RML conversion, i.e. bf:catalogerID --> bf:catalogerID, bf:descriptionLanguage --> bf:descriptionLanguage, etc. There's no admin metadata RT in this section.

The other properties (I'm not sure what they're technically called in JSON-LD) -- "user", "group", "templateId", etc. are added after the RDA-to-BF conversion itself, as part of the last step before uploading to Sinopia. That "templateId" property is the only RT that gets specified in the conversion code -- depending on what the data is, it's WAU:RT:BF2:Works, :Instance, or :Item -- I never select anything else, and don't have an opportunity to. Whichever admin metadata RT nested with :Works, :Instance, or :Item is what gets used to display the admin metadata within that "data" section in Sinopia.

Because bf:catalogerID is in the "data" section but is not displaying, I would think the problem is with the (for this example) WAU:RT:BF2:Works RT, the WAU:RT:BF2:AdminMetadata RT, or Sinopia itself. the :Works and :AdminMetadata RTs look good to me, so I think it's just a Sinopia problem that the devs would have to look at.

Does that make sense?

**Comment by mcm104** [comment]
Just realized I never tagged you in this response @CECSpecialistI !

**Comment by CECSpecialistI** [comment]
Thanks for explaining Melissa! That makes sense now. I can't figure out why this is happening either, my best guess is that it's a Sinopia problem that devs would have to look at, as Melissa says. @mcm104 @gerontakos would either of you like to create an issue in the Sinopia repo, or would you like me to do it?

**Comment by mcm104** [comment]
@CECSpecialistI @gerontakos I can do it!

**Comment by mcm104** [comment]
@CECSpecialistI @gerontakos Just kidding -- as I was writing up the issue, I double-checked the URI for the cataloger ID property to avoid the "sinopa" problem from the last time I raised an issue, and realized that the property is in our code is catalogerID with a capital D, but it's written by BFLC as catalogerId with a lowercase d. It seemed unlikely because the reason it's catalogerID in our BIBFRAME data is because it's catalogerID in our RDA data, and that seems to display the cataloger IDs just fine, but still, worth testing it first before raising an issue.

So I edited the RML to look for catalogerID and output catalogerId -- and bingo! In the test records I've looked at, the cataloger ID now displays in Sinopia!
![catalogerId](https://user-images.githubusercontent.com/50182013/124997378-5a535f80-dfff-11eb-9392-47dbf1ca34be.PNG)

Our BIBFRAME data before:
```
"http://id.loc.gov/ontologies/bflc/catalogerID": [
  {
    "@value": "cec23"
  }
],
```

Our BIBFRAME data now:
```
"http://id.loc.gov/ontologies/bflc/catalogerId": [
  {
    "@value": "cec23"
  }
],
```

However, this begs the question -- why do the cataloger ID names display for our RDA data where the property is written as "catalogerID", when it did not work for many (if not all) of our BIBFRAME records?

Here is an example of it working in RDA: <https://api.sinopia.io/resource/1a09069d-8165-42ae-bfc0-f6b269b4d34b>
```
"http://id.loc.gov/ontologies/bflc/catalogerID": [
  {
    "@value": "cec23",
    "@language": "en"
  }
],
```
So it's written the wrong way (catalogerI**D**), but it displays just fine:
![catalogerID2](https://user-images.githubusercontent.com/50182013/124997667-e82f4a80-dfff-11eb-9211-44266566d203.PNG)

tl;dr, it's fixed, but I'm still not sure why the fix was necessary!

**Comment by gerontakos** [comment]
I don't think we need a Sinopia issue, it looks like our error.
Our instances of bf:Work call WAU:RT:BF2:Works which, in turn, calls WAU:RT:BF2:AdminMetadata which, in turn, uses the property label=Cataloger ID and propertyURI=<http://id.loc.gov/ontologies/bflc/catalogerId>. Our instances of RDA Works use their format-specific templates; so like for a sound recording, the work calls WAU:RT:RDA_Work:sound_recordings which, in turn, calls WAU:RT:BF2:BFLC Admin Metadata which, in turn, uses the property label=Your cataloger ID (UW netid)(*) and propertyURI=<http://id.loc.gov/ontologies/bflc/catalogerID>. So what I see is that the RDA RTs have incorrectly used /catalogerID rather than /catalogerId. I don't think we should fix this, even though, if I'm not mistaken, RFC 3986 specifies that a URI's protocol representation (http://) and host representation are NOT case sensitive, whereas the remainder of the path IS case sensitive. That's in the Sinopia RTs. If we look at the data (as Melissa already showed above) we see:
(1) BF Work:
"http://id.loc.gov/ontologies/bflc/catalogerId": [
        {
          "@value": "lins@uw.edu"
        }
      ].
(2) RDA Work:
"http://id.loc.gov/ontologies/bflc/catalogerID": [
        {
          "@value": "lins@uw.edu",
          "@language": "eng"
        }
      ].
 So I believe it's our mistake -- one we should not bother to fix at this time. Considering that we plan to remodel the RDA admin metadata, it really seems like a waste of time to repair in the RDA data and Sinopia RTs used by the RDA data. However it was important to repair in the BIBFRAME and that's done!
